program/src/instructions/wns/buy_spl.rs

```diff
@@ -351,6 +353,32 @@ pub fn process_buy_wns_spl<'info, 'b>(
         None,
     )?;
 
+    let asset_id = ctx.accounts.mint.key();
+
+    record_event(
+        &TcompEvent::Taker(TakeEvent {
+            taker: *ctx.accounts.buyer.key,
+            bid_id: None,
+            target: Target::AssetId,
+            target_id: asset_id,
+            field: None,
+            field_id: None,
+            amount,
+            quantity: 0, //quantity left
+            tcomp_fee,
+            taker_broker_fee,
+            maker_broker_fee,
+            creator_fee, // Can't record actual because we transfer lamports after we send noop tx
+            currency,
+            asset_id: Some(asset_id),
+        }),
+        &ctx.accounts.marketplace_program,
+        TcompSigner::List(&ctx.accounts.list_state),
+    )?;
+
+    let price = unwrap_checked!({ amount.checked_add(creator_fee) });
```

Comment:
why only creator fee and not other fees? Seems incomplete

Reply:
only one that can be used to rug the user; tensor fees are hard-coded into the contract and if you use a tensor contract you trust Tensor not to rug you by default